### PR TITLE
[layernode] Move activation to LayerNode @open sesame 06/22 18:12

### PR DESCRIPTION
**Pull request by kparichay**
Move activation from Layer internal to LayerNode

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: nntrainer/graph/network_graph.cpp [hunk 1]
```diff
@@ -172,7 +171,8 @@ int NetworkGraph::realizeActivationType(
   }
 
   std::shared_ptr<LayerV1> layer = lnode->getObject();
-  layer->setActivation(act);
+  lnode->setProperty({"activation=" + ActivationTypeStr[(unsigned int)act]});
+  in_node->setProperty({"activation=none"});
```
Comment (jijoongmoon) [inline nntrainer/graph/network_graph.cpp:175]: I wonder that if the activation property is set in layer_node, then does it set the layer property as well?? 
It looks like when we call setProperty of layer node, then setActivation() is called and seems like it does not set the layer property. Is it OK without set the layer property?

**Review by jijoongmoon — APPROVED** [review]
LGTM, except minor comments.

File: nntrainer/graph/network_graph.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/graph/network_graph.cpp:175]: > I wonder that if the activation property is set in layer_node, then does it set the layer property as well??

Yes, the layer property is also set. The throw in setProperty() after calling setActivation() leads to the setting the activation in layer as well (which is the activation layer). The throw method is replaced with a  better method in #1332.

**Review by zhoonit — APPROVED** [review]
lgtm

**Pull request merged**